src/priceHistory/PriceHistoryChart.tsx

```diff
@@ -0,0 +1,363 @@
+
+    // Only grab one price per chart step & the most recent price
+    if (
+      lastTimestampAdded &&
+      priceHistoryPrices[i].priceFetchedAt - lastTimestampAdded <
+        ONE_HOUR_IN_MILLIS * chartStepInHours &&
+      i !== priceHistoryPrices.length - 1
+    ) {
+      continue
+    } else {
+      lastTimestampAdded = priceHistoryPrices[i].priceFetchedAt
+      const { priceUsd } = priceHistoryPrices[i]
+      const localAmount = dollarsToLocal(priceUsd)
+      chartData.push({
+        amount: localAmount ? localAmount.toNumber() : 0,
+        displayValue: localAmount ? displayLocalCurrency(localAmount) : '',
+        priceFetchedAt: priceHistoryPrices[i].priceFetchedAt,
+      })
+    }
+  }
+
+  // Make sure the highest and lowest prices are included in the chart
+  if (highPrice) {
+    chartData.push({
+      amount: dollarsToLocal(highPrice.priceUsd)?.toNumber() || 0,
+      displayValue: displayLocalCurrency(dollarsToLocal(highPrice.priceUsd) || 0),
+      priceFetchedAt: highPrice.priceFetchedAt,
+    })
+  }
+  if (lowPrice) {
+    chartData.push({
+      amount: dollarsToLocal(lowPrice.priceUsd)?.toNumber() || 0,
+      displayValue: displayLocalCurrency(dollarsToLocal(lowPrice.priceUsd) || 0),
+      priceFetchedAt: lowPrice.priceFetchedAt,
+    })
+  }
+
+  const sortedChartData = chartData.sort((a, b) => a.priceFetchedAt - b.priceFetchedAt)
+  const uniqueChartData = uniqBy(sortedChartData, 'priceFetchedAt')
+  return uniqueChartData
+}
+
+interface PriceHistoryChartProps {
+  tokenId: string
+  containerStyle?: ViewStyle
+  testID?: string
+  chartPadding?: number
+  color?: colors
+  step?: number
+}
+
+export default function PriceHistoryChart({
+  tokenId,
+  containerStyle,
+  testID,
+  chartPadding,
+  color = colors.black,
+  step = CHART_STEP_IN_HOURS,
+}: PriceHistoryChartProps) {
+  const dispatch = useDispatch()
+  const localCurrencyCode = useSelector(getLocalCurrencyCode)
+  const localExchangeRate = useSelector(usdToLocalCurrencyRateSelector)
+  const prices = useSelector((state: RootState) => priceHistoryPricesSelector(state, tokenId))
+  const status = useSelector((state: RootState) => priceHistoryStatusSelector(state, tokenId))
+
+  const dollarsToLocal = useCallback(
+    (amount: BigNumber.Value | null) =>
+      convertDollarsToLocalAmount(amount, localCurrencyCode ? localExchangeRate : 1),
+    [localExchangeRate]
+  )
+
+  const displayLocalCurrency = useCallback(
+    (amount: BigNumber.Value) =>
+      getLocalCurrencyDisplayValue(amount, localCurrencyCode || LocalCurrencyCode.USD, true),
+    [localCurrencyCode]
+  )
+
+  useEffect(() => {
+    const latestTimestamp = prices.at(-1)?.priceFetchedAt ?? 0
+    if (prices.length > 0 && latestTimestamp > Date.now() - ONE_HOUR_IN_MILLIS) {
+      return
+    }
+
+    dispatch(
+      fetchPriceHistoryStart({
+        tokenId,
+        startTimestamp: Date.now() - ONE_DAY_IN_MILLIS * 30,
+        endTimestamp: Date.now(),
+      })
+    )
+  }, [])
```

Comment:
should we include tokenId and prices in the dependency array?

Reply:
Fixed in bd1989815b43713eb2834cad17ec59de6d2abacf!